web/tests/functional/authentication/test_permission_management.py

```diff
@@ -0,0 +1,220 @@
+        expected_permission_set = set(
+            [Permission.PRODUCT_ADMIN, Permission.PRODUCT_ACCESS,
+             Permission.PRODUCT_STORE])
+        self.assertSetEqual(
+            expected_permission_set, product_permission_set,
+            "PRODUCT permission set in the database is different.")
+
+    def test_system_permissions(self):
+        sys_permissions = self._root_auth_client.getPermissions('SYSTEM')
+        sys_permission_set = set(sys_permissions)
+        expected_permission_set = set(
+            [Permission.SUPERUSER])
+        self.assertSetEqual(
+            expected_permission_set, sys_permission_set,
+            "SYSTEM permission set in the database is different.")
+
+    def test_user_product_permissions(self):
+        normal_test_user_name = "john"
+        as_user = False
+        expected_permission = Permission.PRODUCT_STORE
+
+        # Check test user if he is in the authorization database with
+        # PRODUCT_STORE right.
+        authorized_names = self._root_auth_client.getAuthorisedNames(
+            expected_permission, self._extra_params_as_json)
+        self.assertIn(normal_test_user_name, authorized_names.users,
+                      "Test database mismatch.")
+
+        # Check test user has not got PRODUCT_ACCESS permission for test
+        # product.
+        expected_permission = Permission.PRODUCT_ACCESS
+        authorized_names = self._root_auth_client.getAuthorisedNames(
+            expected_permission, self._extra_params_as_json)
+        self.assertNotIn(normal_test_user_name, authorized_names.users,
+                         "Test database mismatch.")
+
+        # Add John to the test product with PRODUCT_ACCESS permission.
+        # But his name is in uppercase form.
+        result = self._root_auth_client.addPermission(
+            expected_permission, normal_test_user_name.upper(), as_user,
+            self._extra_params_as_json)
+        self.assertTrue(result)
+
+        # Read back John's permission with his "original" name.
+        authorized_names = self._root_auth_client.getAuthorisedNames(
+            expected_permission, self._extra_params_as_json)
+        self.assertIn(normal_test_user_name, authorized_names.users,
+                      "Could not give permission for user.")
+
+        # Remove permission from John with uppercase name.
+        result = self._root_auth_client.removePermission(
+            expected_permission, normal_test_user_name.upper(), as_user,
+            self._extra_params_as_json)
+        self.assertTrue(result)
+
+        # Check that John really loose his PRODUCT_ACCESS permission.
+        authorized_names = self._root_auth_client.getAuthorisedNames(
+            expected_permission, self._extra_params_as_json)
+        for user in authorized_names.users:
+            self.assertNotEqual(normal_test_user_name.lower(), user.lower())
+
+    def __getRealGroupName(self, group_name_guess):
```

Comment:
It would be better to move private functions after the `setUp` function but before test cases. This way we will not mix test cases and functions which are not test cases.

Also please use lower case function names separated by underscore:
```suggestion
    def __get_real_group_name(self, group_name_guess):
```

Reply:
Done.